### PR TITLE
VV10 Grid fix

### PR DESCRIPTION
**Pull request by dgasmith**
## Description
Fixes an issues where VV10 occasionally seg faults due to improperly sized temporary arrays.

Also, spot the error of why the Naive grid blocker has been seg faulting for years in [#9575ede](https://github.com/dgasmith/psi4/commit/9575ede082c5287be1c176a13299829317b3b719). Highly recommend we stop using raw pointers and C functions.

## Status
- [x] Ready to go

### CODE REVIEW

File: psi4/src/export_functional.cc [hunk 1]
```diff
@@ -261,8 +261,13 @@ void export_functional(py::module &m) {
         .def("blocks", &MolecularGrid::blocks, "Returns a list of blocks.");
 
     py::class_<DFTGrid, std::shared_ptr<DFTGrid>, MolecularGrid>(m, "DFTGrid", "docstring")
-        .def_static("build", [](std::shared_ptr<Molecule> &mol, std::shared_ptr<BasisSet> &basis) {
-            return DFTGrid(mol, basis, Process::environment.options);
+        .def_static("build",
+                    [](std::shared_ptr<Molecule> &mol, std::shared_ptr<BasisSet> &basis) {
+                        return std::shared_ptr<DFTGrid>(new DFTGrid(mol, basis, Process::environment.options));
```
Comment (loriab) [inline psi4/src/export_functional.cc]: I'm sure everyone realizes that I'm a one-trick pony wrt C++ style, but can this be:
`return std::make_shared<DFTGrid>(mol, basis, Process::environment.options);`

File: psi4/src/psi4/libfock/cubature.cc [hunk 2]
```diff
@@ -4261,7 +4263,7 @@ void NaiveGridBlocker::block()
     ::memcpy((void*)y_,(void*)y_ref_, sizeof(double)*npoints_);
     ::memcpy((void*)z_,(void*)z_ref_, sizeof(double)*npoints_);
     ::memcpy((void*)w_,(void*)w_ref_, sizeof(double)*npoints_);
-    ::memcpy((void*)index_,(void*)index_ref_, sizeof(double)*npoints_);
+    ::memcpy((void*)index_,(void*)index_ref_, sizeof(int)*npoints_);
```
Comment (susilehtola) [inline psi4/src/psi4/libfock/cubature.cc]: Nice :D

File: psi4/src/psi4/libfock/cubature.cc [hunk 3]
```diff
@@ -4103,9 +4103,11 @@ void MolecularGrid::block(int max_points, int min_points, double max_radius)
     // Reassign
     std::shared_ptr<GridBlocker> blocker;
     if (options_.get_str("DFT_BLOCK_SCHEME") == "NAIVE") {
-        blocker = std::shared_ptr<GridBlocker>(new NaiveGridBlocker(npoints_,x_,y_,z_,w_,index_,max_points,min_points,max_radius,extents_));
+        blocker = std::shared_ptr<GridBlocker>(
```
Comment (andysim) [inline psi4/src/psi4/libfock/cubature.cc]: Pretty sure `make_shared` works here, which may be cleaner

**Comment by dgasmith** [comment]
Might want to do a `make_shared` overhaul at some later date. I think we can do it all in one go with a Python script.

**Pull request merged**